File: java/src/main/java/gov/doc/ntia/sigmf/examples/location/SpeedAndBearingExample.java
package gov.doc.ntia.sigmf.examples.location;

import gov.doc.ntia.sigmf.Capture;
import gov.doc.ntia.sigmf.Global;
import gov.doc.ntia.sigmf.MetaDoc;
import gov.doc.ntia.sigmf.examples.Example;
import gov.doc.ntia.sigmf.examples.ExampleUtils;
import gov.doc.ntia.sigmf.ext.global.location.CoordinateSystem;
import gov.doc.ntia.sigmf.ext.global.location.Location;
import gov.doc.ntia.sigmf.ext.global.sensor.Sensor;

import java.util.Calendar;
import java.util.Date;
import java.util.List;

public class SpeedAndBearingExample implements Example {
    @Override
    public MetaDoc getExample() {
        Date now = Calendar.getInstance().getTime();
        MetaDoc metadoc = new MetaDoc();

        Global global = ExampleUtils.getGlobal(now);
        List<Capture> captures = ExampleUtils.getCaptures(now);
        metadoc.setGlobal(global);
        metadoc.setCaptures(captures);


        Sensor sensor = new Sensor();
        sensor.setId("Greyhound 10");
        Location location = new Location();
        location.setSpeed(35.0);
        location.setBearing(90.0);
        location.setX(0d);
        location.setY(0d);
        location.setZ(0d);

        CoordinateSystem coordinateSystem = new CoordinateSystem();
        coordinateSystem.setId("SpeedAndBearing_1");
        coordinateSystem.setDescription("System used for relative positioning in tunnels.");
        coordinateSystem.setDistanceUnit("miles");
        coordinateSystem.setTimeUnit("hours");
        coordinateSystem.setOrigin("Tunnel Entrance");
        coordinateSystem.setOrientationRef("Magnetic north");
        coordinateSystem.setElevationUnit("feet");
        coordinateSystem.setElevationRef("Tunnel Entrance");

        metadoc.getGlobal().setCoordinateSystem(coordinateSystem);
        global.setSensor(sensor);


        return metadoc;
    }
}

File: Csharp/sigmf/sigmf/examples/location/SpeedAndBearingExample.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf.examples.location
{
    public class SpeedAndBearingExample : Example
    {
        public override MetaDoc getExample()
        {
            DateTime now = getCurrentTime();
            MetaDoc metaDoc = new MetaDoc();
            Globals globals = ExampleUtils.getGlobals(now);
            List<Capture> captures = ExampleUtils.getCaptures(now);
            metaDoc.setGlobal(globals);
            metaDoc.setCaptures(captures);

            CoordinateSystem coordinateSystem = new CoordinateSystem();
            coordinateSystem.Id = "SpeedAndBearing_1";
            coordinateSystem.Description = "System used for relative positioning in tunnels.";
            coordinateSystem.DistanceUnit = "miles";
            coordinateSystem.TimeUnit = "hours";
            coordinateSystem.Origin = "Tunnel Entrance";
            coordinateSystem.OrientationRef = "Magnetic north";
            coordinateSystem.ElevationUnit = "feet";
            coordinateSystem.ElevationRef = "Tunnel Entrance";

            (metaDoc.getGlobal()).CoordinateSystem = coordinateSystem;

            LocationAnnotation a1 = new LocationAnnotation();
            a1.SampleStart = 01;
            a1.SampleCount = 1001;
            a1.Speed = 3.0;
            a1.Bearing = 89.7;
            a1.Z = -3.23;

            metaDoc.addAnnotation(a1);

            return metaDoc;
        }
        private DateTime getCurrentTime()
        {
            DateTime st = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
            TimeSpan t = (DateTime.Now.ToUniversalTime() - st);
            return st;
        }
    }
}
